src/projected.jl

```diff
@@ -0,0 +1,54 @@
+abstract type AbstractProjectedCoords <: AbstractSkyCoords end
+
+""" Projected coordinates with origin and offset.
+Approximation for small FoV. """
+struct ProjectedCoords{TC<:AbstractSkyCoords,T} <: AbstractProjectedCoords
+    origin::TC
+    offset::SVector{2,T}
+end
+
+""" Projected coordinates with origin (stored in the type) and offset.
+Approximation for small FoV. """
+struct ProjectedCoordsS{C,T} <: AbstractProjectedCoords
+    offset::SVector{2,T}
+end
+
+ProjectedCoordsS{C}(offset) where {C} = ProjectedCoordsS{C, eltype(offset)}(offset)
+constructorof(::Type{<:ProjectedCoordsS{C}}) where {C} = ProjectedCoordsS{C} ∘ tuple
+
+parent_coords_type(::Type{ProjectedCoords}) = Any
+parent_coords_type(::Type{ProjectedCoordsS}) = Any
+parent_coords_type(::Type{<:ProjectedCoords{TC}}) where {TC} = TC
+parent_coords_type(::Type{<:ProjectedCoordsS{TC}}) where {TC <: AbstractSkyCoords} = TC
+parent_coords_type(::Type{<:ProjectedCoordsS{origin}}) where {origin} = typeof(origin)
```

Comment:
I'm not quite following why we need this- not used locally and is quite noisy- I would prefer if the interface could be used in a simpler way like

```julia
proj = ProjectedCoord() #...
parent_type = typeof(parent(proj))
```

this is the "obvious" way I would expect this to work.

Reply:
I still think this is a bit of an unnatural piece of API. Why not

```julia
C = typeof(origin(proj))
```

or, for example

```julia
function (proj::ProjectedCoords{C}) where C
	# ...
end
```